src/misc/matching.c

```diff
@@ -1007,6 +1007,450 @@ static igraph_error_t igraph_i_maximum_bipartite_matching_weighted(
+
+                                // unflag finished
+                                finished = false;
+
+                                // B is blossom formed by e and edges on the path v -> w in T
+                                // find by traversing back v -> root(v) blossom marking each vertex,
+                                // then traverse w -> root(w) and first blossom marked is blossom root
+                                // then continue to root(w), removing marks instead
+                                current = v;
+                                VECTOR(b_mark)[current] = true;
+                                while (current != -1 && !VECTOR(f_is_root)[current]) {
+                                    current = VECTOR(f_parent)[current];
+                                    VECTOR(b_mark)[current] = true;
+                                }
+                                current = w;
+                                blossom_root = -1;
+                                if (VECTOR(b_mark)[current]) {
+                                    blossom_root = current;
+                                    debug("found blossom root\n");
+                                } else {
+                                    VECTOR(b_mark)[current] = true;
+                                }
+                                while (!VECTOR(f_is_root)[current]) {
+                                    current = VECTOR(f_parent)[current];
+                                    if (blossom_root == -1) {
+                                        if (VECTOR(b_mark)[current]) {
+                                            blossom_root = current;
+                                            debug("found blossom root\n");
+                                        } else {
+                                            VECTOR(b_mark)[current] = true;
+                                        }
+                                    } else {
+                                        VECTOR(b_mark)[current] = false;
+                                    }
+                                }
+                                igraph_vector_int_t *list = igraph_vector_int_list_get_ptr(&b_lists, blossom_root);
+
+                                // mark all vertices in B as contracted to blossom root
+                                // do by traversing again, but stopping when blossom marks stop
+                                current = v;
+                                while (current != -1 && VECTOR(b_mark)[current]) {
+                                    VECTOR(contraction)[current] = blossom_root;
+                                    VECTOR(b_mark)[current] = false;
+                                    igraph_vector_int_push_back(list, current);
+                                    current = VECTOR(f_parent)[current];
+                                }
+                                current = w;
+                                igraph_vector_int_resize(&temp_list, 0);
```

Comment:
Generally, all functions that return an error code must be protected using `IGRAPH_CHECK`.